### PR TITLE
Add filter docs and make some small updates

### LINKED ISSUE #98: Update docs for postgresql
Our README and hosted documentation still contain references to mongo (and no references to postgresql). 

We should update the README and getting started guide with instructions about deploying PSQL.

### PR DESCRIPTION
**Pull request by john-dupuy**
* Adding documentation about filters
* Include links to filter docs in UI
* Update OKD deploy template in documentation
* Remove references to MongoDB in documentation
* Make README.md point to https://docs.ibutsu-project.org/en/latest/, no need to manage two sources of documentation.
* Improve descriptions of parameters when creating widgets.
* Move `additional_filters` into a common parameter `_ADDITIONAL_FILTERS_PARAM` in `constants.py`

depends on #106 

Should close #98

### CODE REVIEW

File: docs/source/user-guide/filter-help.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     - regex
+     - ``Result.<column>.op("~")(<value>)``
+     - Regex match that ``column`` contains ``value``
+   * - @
+     - exists
+     - ``Result.<column> != None`` or ``Result.<column> == None``
+     - Column exists and is defined
+
+The information for this table is taken from
+`filters.py <https://github.com/ibutsu/ibutsu-server/blob/master/backend/ibutsu_server/filters.py>`_.
```
Comment (rsnyman) [inline docs/source/user-guide/filter-help.rst:54]: A nicer/cleaner way of making links is by using references. It's almost identical to this, except that the links are at the bottom of the document. For example:

```
The information for this table is taken from `filters.py`_

(rest of the document)

.. _filters.py: https://github.com/ibutsu/ibutsu-server/blob/master/backend/ibutsu_server/filters.py
```

This way, if you want to specify a link more than once, you can just use the same reference in all the places.

**Review by rsnyman — CHANGES_REQUESTED** [review]
Overall, this is fantastic. One change you need to make is in the getting started guide, and I have one small suggestion on linking.

File: docs/source/user-guide/filter-help.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+These are simple text input components that take a string of comma-separated filters. Here, a user
+is meant to use the ``operator`` in the table above to filter on the appropriate column. Some examples
+of these filters include:
+
+* ``metadata.tags=platform-experience,source=local,summary.passes>10``
+* ``metadata.assignee=jdoe,metadata.exception_name=AssertionError,result=failed``
+
+.. _filters.py: https://github.com/ibutsu/ibutsu-server/blob/master/backend/ibutsu_server/filters.py
+.. _Runs/Results: https://github.com/ibutsu/ibutsu-server/blob/master/backend/ibutsu_server/db/models.p
```
Comment (rsnyman) [inline docs/source/user-guide/filter-help.rst:91]: I think you left a "y" off the end of this.

File: docs/source/user-guide/filter-help.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     - regex
+     - ``Result.<column>.op("~")(<value>)``
+     - Regex match that ``column`` contains ``value``
+   * - @
+     - exists
+     - ``Result.<column> != None`` or ``Result.<column> == None``
+     - Column exists and is defined
+
+The information for this table is taken from
+`filters.py`_.
```
Comment (rsnyman) [inline docs/source/user-guide/filter-help.rst:54]: While you're fixing the issue below, just pop this on the previous line? It's so lonely here on this line by itself...

**Review by rsnyman — CHANGES_REQUESTED** [review]

File: docs/source/user-guide/filter-help.rst (hunk 3 above)
Comment (john-dupuy) [reply docs/source/user-guide/filter-help.rst:54]: words get lonely too sometimes

**Review by rsnyman — APPROVED** [review]

**Pull request merged**